src/main/java/com/google/sps/data/ComputeAPIClient.java

```diff
@@ -0,0 +1,304 @@
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.lang.reflect.*;
+
+/** Class that creates, stops and deletes VM instances. */
+public class ComputeAPIClient implements ComputeAPIClientInterface {
+  // Constants
+  private static final String APPLICATION_NAME = "VMN";
+  private static final String PROJECT_ID = "cdesir-step-2020";
+  private static final String ZONE_NAME = "us-central1-f";
+  private static final String SOURCE_IMAGE_PREFIX =
+    "https://www.googleapis.com/compute/v1/projects/";
+  private static final String SOURCE_IMAGE_PATH =
+    "cdesir-step-2020/global/images/vmn-image";
+  private static final String NETWORK_INTERFACE_CONFIG = "ONE_TO_ONE_NAT";
+  private static final String NETWORK_ACCESS_CONFIG = "External NAT";
+  private static final long OPERATION_TIMEOUT_MILLIS = 60 * 1000;
+  private static final JsonFactory JSON_FACTORY = 
+    JacksonFactory.getDefaultInstance();
+  private final String OPERATION_STATUS_DONE = "DONE";
+  private final String VNC_SERVER_PASSWORD_KEY = "vnc-server-password";
+  private final String STARTUP_SCRIPT_URL_KEY = "startup-script-url";
+  private final String SCOPE_FULL_CONTROL = 
+    "https://www.googleapis.com/auth/devstorage.full_control";
+  private final String SCOPE_COMPUTE = 
+    "https://www.googleapis.com/auth/compute";
+  // Member variables 
+  private static GoogleCredentials credential;
+  private Compute compute;
+  private static HttpTransport httpTransport;  
+
+  /** Create environment for Compute Engine object. */
+  public ComputeAPIClient() {
+    createComputeEngineObject();
+  }
+  
+  /** 
+   * Create Compute Engine object and authenticate using Google Application
+   * Default Credentials
+   */
+  private void createComputeEngineObject() {
+    try {
+      httpTransport = GoogleNetHttpTransport.newTrustedTransport();
+      credential = GoogleCredentials.getApplicationDefault();
+      if (credential.createScopedRequired()) {
+        List<String> scopes = new ArrayList<>();
+        scopes.add(ComputeScopes.DEVSTORAGE_FULL_CONTROL);
+        scopes.add(ComputeScopes.COMPUTE);
+        credential = credential.createScoped(scopes);
+      }
+      HttpRequestInitializer requestInitializer = 
+        new HttpCredentialsAdapter(credential);
+      // Create Compute Engine object.
+      compute = new Compute.Builder
+        (httpTransport, JSON_FACTORY, requestInitializer)
+        .setApplicationName(APPLICATION_NAME)
+        .build();
+    } catch (IOException e) {
+        System.out.println("Got error!");
+        System.err.println(e.getMessage());
+    } catch (Throwable t) {
+        t.printStackTrace();
+      }
+  }
+
+  public void createInstance(String instanceName, String vncServerPassword)
+    throws IOException {
+      Instance instance = new Instance();
+      instance.setName(instanceName);
+      instance.setMachineType(String.format(
+        "projects/%s/zones/%s/machineTypes/n1-standard-1",
```

Comment:
Create a constant for the machine type:
private static final String MACHINE_TYPE = "n1-standard-1";
And then here you can do:
("projects/%s/zones/%s/machineTypes/%s", ...ZONE_NAME,MACHINE_TYPE)